src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -63,22 +65,41 @@ public function process(ContainerBuilder $container)
         $definition = $container->findDefinition($this->dispatcherService);
 
         foreach ($container->findTaggedServiceIds($this->listenerTag, true) as $id => $events) {
+            $reflection = null;
             foreach ($events as $event) {
                 $priority = isset($event['priority']) ? $event['priority'] : 0;
 
                 if (!isset($event['event'])) {
-                    throw new InvalidArgumentException(sprintf('Service "%s" must define the "event" attribute on "%s" tags.', $id, $this->listenerTag));
+                    if (null === $reflection) {
+                        $class = $container->getDefinition($id)->getClass();
+                        if (!$reflection = $container->getReflectionClass($class)) {
+                            throw new InvalidArgumentException(sprintf('Class "%s" used for service "%s" cannot be found.', $class, $id));
+                        }
+                    }
+                    if (!$reflection->implementsInterface(EventListenerInterface::class)) {
+                        throw new InvalidArgumentException(sprintf('Service "%s" must define the "event" attribute on "%s" tags or implements the "%s" interface.', $id, $this->listenerTag, EventListenerInterface::class));
+                    }
+
+                    $event['event'] = $this->guessListenedClass($reflection, $id);
+                    $event['method'] = '__invoke';
+                } else {
+                    $event['event'] = $aliases[$event['event']] ?? $event['event'];
                 }
-                $event['event'] = $aliases[$event['event']] ?? $event['event'];
 
                 if (!isset($event['method'])) {
                     $event['method'] = 'on'.preg_replace_callback([
                         '/(?<=\b)[a-z]/i',
                         '/[^a-z0-9]/i',
-                    ], function ($matches) { return strtoupper($matches[0]); }, $event['event']);
+                    ], function ($matches) { return strtoupper($matches[0]); }, (false === $p = \strrpos($event['event'], '\\')) ? $event['event'] : \substr($event['event'], $p + 1));
```

Comment:
the `\` should be removed, we use it only on a short list of special functions

Reply:
done